File: 3rd-SEM/AD_Lab/Assignment_8/Stack.java
public class Stack {
    int stack[];
    int top;
    int size;

    public Stack(int size) {
        this.size = size;
        stack = new int[size];
        top = -1;
    }

    public void push(int value) {
        if (top == size - 1) {
            System.out.println("Stack Overflow");
        } else {
            stack[++top] = value;
            System.out.println(value + " pushed to stack");
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public void pop() {
        if (isEmpty()) {
            System.out.println("Stack Underflow");
        } else {
            System.out.println("Element popped: " + stack[top--]);
        }
    }

    public int top() {
        if (isEmpty()) {
            System.out.println("Stack Underflow");
            return -1;
        }
        return stack[top];
    }

    public void size() {
        System.out.println("Size: " + (top + 1));
    }

    public static void main(String[] args) {
        Stack s = new Stack(5);
        s.push(30);
        s.push(12);
        s.push(5);
        s.pop();
        s.size();
    }
}

File: 5th-SEM/C/Misc/stack.c
#include <stdio.h>
#define N 5
int top= -1;
int stack[N];
void push(int n){
    if (top== N-1){
        printf("Stack Overflow\n");
    }else{
        top++;
        stack[top]=n;
    }
}

void pop(){
    int item;
    if(top==-1){
        printf("Underflow\n");
    }else{
        item=stack[top];
        top--;
    }
    printf("%d popped out\n",item);
}

int peek(){
    if(top==-1){
        printf("Stack is Empty\n");
    }else{
        return stack[top];
    }
}
void display(){
    for(int i=top;i>=0;i--){
        printf("%d \n",stack[i]);
    }
}
int main(void) {
    push(7);
    push(5);
    push(2);
    pop();
    display();
    return 0;
}
